### PR TITLE
Личный проект: шаблонизируй это

### PR DESCRIPTION
**Pull request by katasonova**
---
:mortar_board: [Личный проект: шаблонизируй это](https://up.htmlacademy.ru/ecmascript/10/user/579401/tasks/7)

:boom: https://htmlacademy-ecmascript.github.io/579401-cinemaddict-10/5/

### CODE REVIEW

File: src/components/extra-movie-details.js [hunk 1]
```diff
@@ -1,4 +1,15 @@
-export const renderExtraMovieDetails = () => {
+const renderGenres = (genres) => {
+  if (typeof genres === `string`) {
```
Comment (fyvfyv) [inline src/components/extra-movie-details.js]: Can `genres` be not a string?

File: src/mock/menu.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+const getWatchlistNumber = (cards) => {
+  let number = 0;
+  cards.forEach((card) => card.isInWatchlist ? number++ : number);
+  return number;
```
Comment (fyvfyv) [inline src/mock/menu.js]: You can do it with `filter` method in one line

File: src/mock/movie-card.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import {getRandomInteger} from '../utils';
+
+const MAX_RATING = 10;
+const DESCRIPTION = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui. Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`;
+const MAX_SENTENCES = 3;
+
+const Title = [
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: `titles`

File: src/mock/movie-card.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+};
+
+const Months = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];
+
+const Days = {
+  MIN: 1,
+  MAX: 28
+};
+
+const Genre = [`Action`, `Adventure`, `Comedy`, `Crime`, `Drama`, `Fantasy`, `Historical`, `Thriller`];
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: `genres`

File: src/mock/movie-card.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    MIN: 1,
+    MAX: 3
+  },
+  MIN: {
+    MIN: 1,
+    MAX: 59
+  }
+};
+
+const Months = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: `months`

File: src/mock/movie-card.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+const getMovieDuration = () => {
+  return `${getRandomInteger(MovieDuration.HOURS.MIN, MovieDuration.HOURS.MAX)}h ${getRandomInteger(MovieDuration.MIN.MIN, MovieDuration.MIN.MAX)}m`;
+};
+
+const getRandomBoolean = () => Math.random() >= 0.5;
+
+const getMovieTitle = () => getRandomArrayElement(Title);
+const getYear = () => getRandomInteger(Year.MIN, Year.MAX);
+
+const getRandomArrayElements = (array, numberOfElements) => array.filter(() => Math.random() > 0.5).slice(0, numberOfElements);
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: You have func `getRandomBoolean`

File: src/mock/movie-card.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+const getRandomDescription = (string) => {
+  return string.split(`.`).filter(() => Math.random() > 0.5).slice(0, Math.ceil(Math.random() * MAX_SENTENCES)).join(`.`);
+};
+
+const getMovieDuration = () => {
+  return `${getRandomInteger(MovieDuration.HOURS.MIN, MovieDuration.HOURS.MAX)}h ${getRandomInteger(MovieDuration.MIN.MIN, MovieDuration.MIN.MAX)}m`;
+};
+
+const getRandomBoolean = () => Math.random() >= 0.5;
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: to `utils.js`

File: src/mock/movie-card.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+  `Itly`,
+  `Spain`
+];
+
+const getRandomArrayElement = (array) => array[Math.floor(Math.random() * array.length)];
+
+const getRandomRating = () => (Math.random() * MAX_RATING).toFixed(1);
+
+const getRandomDescription = (string) => {
+  return string.split(`.`).filter(() => Math.random() > 0.5).slice(0, Math.ceil(Math.random() * MAX_SENTENCES)).join(`.`);
```
Comment (fyvfyv) [inline src/mock/movie-card.js]: You have func `getRandomBoolean`

File: src/components/extra-movie-details.js (hunk 1 above)
Comment (katasonova) [reply src/components/extra-movie-details.js]: yes, if it's not a string a function will jump straight away to the line 5

Comment (fyvfyv) [reply src/components/extra-movie-details.js]: But how `genres` can be different types, if your function in mocks create array every time?

Comment (katasonova) [reply src/components/extra-movie-details.js]: gotcha, it was my first draft, cheers :)

**Pull request merged**